server/cromwell.py

```diff
@@ -94,165 +93,160 @@ async def cromwell(request):  # pylint: disable=too-many-locals
-            backend = hb.ServiceBackend(
-                billing_project=dataset,
-                bucket=hail_bucket,
-                token=hail_token,
-            )
-
-            commit = params['commit']
-            if not commit or commit == 'HEAD':
-                raise web.HTTPBadRequest(reason='Invalid commit parameter')
-
-            libs = params.get('dependencies')
-            if not isinstance(libs, list):
-                raise web.HTTPBadRequest(reason='Expected "dependencies" to be a list')
-            cwd = params['cwd']
-            wf = params['workflow']
-            if not wf:
-                raise web.HTTPBadRequest(reason='Invalid script parameter')
-
-            input_jsons = params.get('input_json_paths') or []
-            input_dict = params.get('inputs_dict')
-
-            # This metadata dictionary gets stored at the output_dir location.
-            timestamp = datetime.now().astimezone().isoformat()
-            metadata = json.dumps(
-                get_analysis_runner_metadata(
-                    timestamp=timestamp,
-                    dataset=dataset,
-                    user=email,
-                    access_level=access_level,
-                    repo=repo,
-                    commit=commit,
-                    script=wf,
-                    description=params['description'],
-                    output_suffix=workflow_output_dir,
-                    driver_image=DRIVER_IMAGE,
-                    cwd=cwd,
-                    mode='cromwell',
-                )
-            )
-
-            # Publish the metadata to Pub/Sub. Wait for the result before running the batch.
-            publisher.publish(PUBSUB_TOPIC, metadata.encode('utf-8')).result()
 
-            user_name = email.split('@')[0]
-            batch_name = f'{user_name} {repo}:{commit}/cromwell/{wf}'
-
-            batch = hb.Batch(
-                backend=backend, name=batch_name, requester_pays_project=project
+        server_config = get_server_config()
+        output_dir = validate_output_dir(params['output'])
+        dataset = params['dataset']
+        check_dataset_and_group(server_config, dataset, email)
+        repo = params['repo']
+        check_allowed_repos(server_config, dataset, repo)
+        access_level = params['accessLevel']
+        labels = params.get('labels')
+
+        ds_config = server_config[dataset]
+        project = ds_config.get('projectId')
+        hail_token = ds_config.get(f'{access_level}Token')
+        service_account_json = get_cromwell_key(
+            dataset=dataset, access_level=access_level
+        )
+        # use the email specified by the service_account_json again
+        service_account_dict = json.loads(service_account_json)
+        service_account_email = service_account_dict.get('client_email')
+        if not service_account_email:
+            raise web.HTTPServerError(
+                reason="The service_account didn't contain an entry for client_email"
```

Comment:
```suggestion
                reason='The service_account didn't contain an entry for client_email'
```

Reply:
This has a single quote in the string “didn’t”, it could be escaped but I think black would resolve it back to double quotes and pylint is happy.